build.sh

```diff
@@ -1,101 +1,77 @@
-        exit 1
+DOTNET_INSTALLED_VERSION=$(dotnet --version 2>&1)
+
+if [ "$DOTNET_VERSION" != "$DOTNET_INSTALLED_VERSION" ]; then
+    echo "Installing .NET CLI..."
+    if [ ! -d "$SCRIPT_DIR/.dotnet" ]; then
+      mkdir "$SCRIPT_DIR/.dotnet"
     fi
+    curl -Lsfo "$SCRIPT_DIR/.dotnet/dotnet-install.sh" https://dot.net/v1/dotnet-install.sh
+    bash "$SCRIPT_DIR/.dotnet/dotnet-install.sh" --channel 1.1 --install-dir .dotnet --no-path
+    bash "$SCRIPT_DIR/.dotnet/dotnet-install.sh" --channel 2.1 --install-dir .dotnet --no-path
+    bash "$SCRIPT_DIR/.dotnet/dotnet-install.sh" --version $DOTNET_VERSION --install-dir .dotnet --no-path
+    export PATH="$SCRIPT_DIR/.dotnet":$PATH
+    export DOTNET_ROOT="$SCRIPT_DIR/.dotnet"
 fi
 
-# Restore tools from NuGet.
-pushd "$TOOLS_DIR" >/dev/null
-if [ ! -f $PACKAGES_CONFIG_MD5 ] || [ "$( cat $PACKAGES_CONFIG_MD5 | sed 's/\r$//' )" != "$( $MD5_EXE $PACKAGES_CONFIG | awk '{ print $1 }' )" ]; then
-    find . -type d ! -name . | xargs rm -rf
-fi
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+export DOTNET_SYSTEM_NET_HTTP_USESOCKETSHTTPHANDLER=0
 
-mono "$NUGET_EXE" install -ExcludeVersion
-if [ $? -ne 0 ]; then
-    echo "Could not restore NuGet packages."
-    exit 1
-fi
 
-$MD5_EXE $PACKAGES_CONFIG | awk '{ print $1 }' >| $PACKAGES_CONFIG_MD5
+###########################################################################
+# INSTALL CAKE
+###########################################################################
 
-popd >/dev/null
+CAKE_INSTALLED_VERSION=$(dotnet-cake --version 2>&1)
 
-# Make sure that Cake has been installed.
-if [ ! -f "$CAKE_EXE" ]; then
-    echo "Could not find Cake.exe at '$CAKE_EXE'."
-    exit 1
+if [ "$CAKE_VERSION" != "$CAKE_INSTALLED_VERSION" ]; then
+    if [ ! -f "$CAKE_EXE" ] || [ ! -d "$CAKE_PATH" ]; then
+        if [ -f "$CAKE_EXE" ]; then
+            dotnet tool uninstall --tool-path $TOOLS_DIR Cake.Tool
+        fi
+
+        echo "Installing Cake $CAKE_VERSION..."
+        dotnet tool install --tool-path $TOOLS_DIR --version $CAKE_VERSION Cake.Tool
+        if [ $? -ne 0 ]; then
+            echo "An error occured while installing Cake."
+            exit 1
+        fi
+    fi
+
+    # Make sure that Cake has been installed.
+    if [ ! -f "$CAKE_EXE" ]; then
+        echo "Could not find Cake.exe at '$CAKE_EXE'."
+        exit 1
+    fi
+else
+    CAKE_EXE="dotnet-cake"
 fi
 
+###########################################################################
+# RUN BUILD SCRIPT
+###########################################################################
+
 # Start Cake
-if $SHOW_VERSION; then
-    exec mono "$CAKE_EXE" -version
-else
-    exec mono "$CAKE_EXE" $SCRIPT -verbosity=$VERBOSITY -configuration=$CONFIGURATION -target=$TARGET $DRYRUN "${SCRIPT_ARGUMENTS[@]}"
-fi
\ No newline at end of file
+(exec "$CAKE_EXE" build.cake --bootstrap) && (exec "$CAKE_EXE" build.cake "$@")
```

Comment:
We don't want to require mono like Cake did previously and use the modern pattern of using Cake as a .NET tool cross platform. Need to bootstrap first to allow for the `dotnet` directive to work in the `build.cake`.